core/services/vrf/v2/bhs_feeder_test.go

```diff
@@ -0,0 +1,145 @@
+		bhsKey := cltest.MustGenerateRandomKey(t)
+		bhsKeyAddresses = append(bhsKeyAddresses, bhsKey.Address.String())
+		keys = append(keys, bhsKey)
+		keySpecificOverrides = append(keySpecificOverrides, toml.KeySpecific{
+			Key:          ptr(bhsKey.EIP55Address),
+			GasEstimator: toml.KeySpecificGasEstimator{PriceMax: gasLanePriceWei},
+		})
+		sendEth(t, ownerKey, uni.backend, bhsKey.Address, 10)
+	}
+	keySpecificOverrides = append(keySpecificOverrides, toml.KeySpecific{
+		// Gas lane.
+		Key:          ptr(vrfKey.EIP55Address),
+		GasEstimator: toml.KeySpecificGasEstimator{PriceMax: gasLanePriceWei},
+	})
+
+	keys = append(keys, ownerKey, vrfKey)
+
+	config, _ := heavyweight.FullTestDBV2(t, "vrfv2_needs_blockhash_store", func(c *chainlink.Config, s *chainlink.Secrets) {
+		simulatedOverrides(t, gasLanePriceWei, keySpecificOverrides...)(c, s)
+		c.EVM[0].MinIncomingConfirmations = ptr[uint32](2)
+		c.Feature.LogPoller = ptr(true)
+		c.EVM[0].FinalityDepth = ptr[uint32](2)
+		c.EVM[0].GasEstimator.LimitDefault = ptr(uint32(gasLimit))
+		c.EVM[0].LogPollInterval = models.MustNewDuration(time.Second)
+	})
+
+	heartbeatPeriod := 5 * time.Second
+	// lggr, logs := logger.TestLoggerObserved(t, zapcore.DebugLevel)
+
+	t.Run("bhs_feeder_startheartbeats_happy_path", func(tt *testing.T) {
+		// consumerContracts := uni.consumerContracts
+		// consumerContractAddresses := uni.consumerContractAddresses
+		coordinator := uni.rootContract
+		coordinatorAddress := uni.rootContractAddress
+		batchCoordinatorAddress := uni.batchCoordinatorContractAddress
+		vrfOwnerAddress := ptr(uni.vrfOwnerAddress)
+		vrfVersion := vrfcommon.V2
+
+		app := cltest.NewApplicationWithConfigV2AndKeyOnSimulatedBlockchain(t, config, uni.backend, keys...)
+		require.NoError(t, app.Start(testutils.Context(t)))
+
+		// Create VRF job.
+		vrfJobs := createVRFJobs(
+			t,
+			[][]ethkey.KeyV2{{vrfKey}},
+			app,
+			coordinator,
+			coordinatorAddress,
+			batchCoordinatorAddress,
+			uni.coordinatorV2UniverseCommon,
+			vrfOwnerAddress,
+			vrfVersion,
+			false,
+			gasLanePriceWei)
+		_ = vrfJobs[0].VRFSpec.PublicKey.MustHash()
+
+		var (
+			v2CoordinatorAddress     string
+			v2PlusCoordinatorAddress string
+		)
+
+		if vrfVersion == vrfcommon.V2 {
+			v2CoordinatorAddress = coordinatorAddress.String()
+		} else if vrfVersion == vrfcommon.V2Plus {
+			v2PlusCoordinatorAddress = coordinatorAddress.String()
+		}
+
+		_ = vrftesthelpers.CreateAndStartBHSJob(
+			t, bhsKeyAddresses, app, uni.bhsContractAddress.String(), "",
+			v2CoordinatorAddress, v2PlusCoordinatorAddress, "", 0, 200, heartbeatPeriod)
+
+		// Ensure log poller is ready and has all logs.
+		require.NoError(t, app.GetRelayers().LegacyEVMChains().Slice()[0].LogPoller().Ready())
+		require.NoError(t, app.GetRelayers().LegacyEVMChains().Slice()[0].LogPoller().Replay(testutils.Context(t), 1))
+
+		initTxns := 260
+		// Wait 260 blocks.
+		for i := 0; i < initTxns+1; i++ {
+			uni.backend.Commit()
+		}
```

Comment:
I think this waits 261 blocks, is that the intention?

Reply:
I just needed some number of blocks and then check for `blocktip - 256`. Updated the code

Reply:
Removed the +1